DummyFile.java

```diff
@@ -1,4 +1,40 @@
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.AmazonKinesisClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadRequest;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+
+import example.foo.FooClient;
+
 public class DummyFile {
-    jkgfldksghkjfdhgkfdgkfdhsjnvjkfdnvkjfdnvkjf
-        gfdgdfjkgkdj
+    private AmazonS3 s3Client = AmazonS3ClientBuilder.standard().build();
+    private AmazonKinesisClient kinesisClient = AmazonKinesisClientBuilder.defaultClient();
+    private FooClient fooClient;
+
+    // Should flag and suggest v2
+    public void s3ListObjects() {
+        s3Client.listObjects(new ListObjectsRequest());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).